### DISCUSSION TITLE
Efficient way to use subspace

### DISCUSSION THREAD

**Discussion by happyych** [fumitoh/modelx#143] [Q&A]
Hello, I’m curious about how to efficiently utilize subspaces. I would like to generate results for five different lapse rate scenarios.

#103 Based on your question, I modeled the variables referencing other subspaces using _space.parent[i]. This approach nicely generated the values for the five scenarios across five subspaces.

However, since I’m running a somewhat heavy computation, this method seems to be similar in terms of speed and memory usage to running the five scenarios separately.

My model has over 400 variables, and since the variables that are not affected by lapse rate shocks should maintain their values from the initial run, I want to update only the changed variables. Do you have any recommendations for this? What I ultimately want is the total premium income for the 5 lapse rate × shock value combinations, and if there’s a faster way to calculate this, I’d love to know.

Thanks as always!

**Comment by fumitoh** [comment]
I think [the IntegratedLife model in lifelib](https://lifelib.io/libraries/appliedlife/IntegratedLife.html#module-appliedlife.IntegratedLife) is a good example how to split logic into multiple spaces. Input data, such as motality tables, model point tables, assumption tables(except for those that vary by sensitivity scenarios) are in spaces that are separate from the projection space. So they are executed only once.

**Reply by happyych** [reply]
Sorry for the late reply and thank you for your advice!

I will try to catch up new library you metioned. It seems that works as what I intended.
Thanks a lot !
